src/main/core/manager.c

```diff
@@ -562,15 +557,47 @@ guint32 manager_getNodeBandwidthDown(Manager* manager, GQuark nodeID, in_addr_t
     return networkinterface_getSpeedDownKiBps(interface);
 }
 
-gdouble manager_getLatency(Manager* manager, GQuark sourceNodeID, GQuark destinationNodeID) {
+gdouble manager_getLatencyForAddresses(Manager* manager, Address* sourceAddress,
+                                       Address* destinationAddress) {
     MAGIC_ASSERT(manager);
-    Host* sourceNode = _manager_getHost(manager, sourceNodeID);
-    Host* destinationNode = _manager_getHost(manager, destinationNodeID);
-    Address* sourceAddress = host_getDefaultAddress(sourceNode);
-    Address* destinationAddress = host_getDefaultAddress(destinationNode);
     return controller_getLatency(manager->controller, sourceAddress, destinationAddress);
 }
 
+gdouble manager_getLatency(Manager* manager, GQuark sourceHostID, GQuark destinationHostID) {
+    MAGIC_ASSERT(manager);
+    Host* sourceHost = _manager_getHost(manager, sourceHostID);
+    Host* destinationHost = _manager_getHost(manager, destinationHostID);
```

Comment:
These are already covered by asserts within the other functions (`_manager_getHost`, `host_lookupInterface`, `host_getDefaultAddress`, etc). I think most of the code only does an assert before it's about to dereference the pointer, so that way you don't need to check if it's null at every function call. I'll remove the `MAGIC_ASSERT(manager)` in these `manager_getLatency`, `manager_getReliability`, etc functions since I don't think I should have added them in the first place since we don't ever dereference the manager in these functions.